mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala

```diff
@@ -108,6 +111,21 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     val kmeans = new KMeans()
     testEstimatorAndModelReadWrite(kmeans, dataset, KMeansSuite.allParamSettings, checkModelData)
   }
+
+  test("Initialize using given cluster centers") {
+    val kmeans = new KMeans()
+      .setK(k)
+      .setSeed(1)
+      .setMaxIter(1000)  // Set a fairly high maxIter to make sure the model is converged.
```

Comment:
This sounds expensive.  What if instead we did this:
* a = KMeans with 1 iteration
* b = KMeans with 2 iterations
* c = KMeans with initialModel a, with 1 iteration
* ensure b and c are equal